addon/adapters/pouch.js

```diff
@@ -131,7 +133,8 @@ export default DS.RESTAdapter.extend({
 
   _recordToData: function (store, type, record) {
     var data = {};
-    var serializer = store.serializerFor(type.typeKey);
+    var recordTypeName = this.getRecordTypeName(type);
+    var serializer = store.serializerFor(recordTypeName);
```

Comment:
Thinking about this some more — these uses are conceptually different from `recordTypeName`, so it's probably best not to use it here. E.g., if you override `getRecordTypeName` to provide your own scheme, the serializer is still going to return the data with a camelized key. And if the serializer changes in the future, it shouldn't affect the type name used in the record IDs. So I'll change this.